OgreMain/include/OgrePass.h

```diff
@@ -168,6 +168,9 @@ namespace Ogre {
         bool mPointAttenuationEnabled : 1;
         mutable bool mContentTypeLookupBuilt : 1;
 
+        /// Channels enabled for writing.
+        bool mColourMask[4];
```

Comment:
instead of introducing a new property, which duplicates `mColourWrite`, you should rather extend `mColourWrite` as
`mColourWriteR`, `mColourWriteG` .. Please use the `: 1;` syntax instead of arrays here to keep the size of `Pass` small.

This will avoid inconsistencies like e.g. `mColourWrite == 0 && mColourMask[2] == 1` on the conceptual level.

Consequently extend `colour_write` in scripts with an extended 4 channel format, where the current format would set all 4 channels at once.